source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -171,6 +174,18 @@ UdpProxyFilter::ActiveSession::ActiveSession(ClusterInfo& cluster,
       .connections()
       .inc();
 
+  if (use_original_src_ip_) {
+    const Network::Socket::OptionsSharedPtr socket_options =
+        Network::SocketOptionFactory::buildIpTransparentOptions();
+    const bool ok = Network::Socket::applyOptions(
+        socket_options, *socket_, envoy::config::core::v3::SocketOption::STATE_PREBIND);
+
+    ENVOY_LOG(debug, "The original src is enabled for address {}.",
+              addresses_.peer_->asStringView());
+    socket_->addOptions(socket_options);
```

Comment:
I think that it would be useful call the addOptions for actual socket because we can know what options are actually added.
I know that added options are not checked by my patch now but it will be useful for further usages.
Please give your opinion if it really not needed.